napari_clusters_plotter/_utilities.py

```diff
@@ -93,8 +92,34 @@ def generate_cluster_image(label_image, predictionlist):
     predictionlist_new = np.array(predictionlist) + 1
     predictionlist_new = np.insert(predictionlist_new, 0, 0)
 
+    import importlib
+
+    loader = importlib.find_loader("pyclesperanto_prototype")
+    found = loader is not None
+
+    if found:
+        return relabel_image_cle(label_image, predictionlist_new)
+    else:
+        return relabel_image_numpy(label_image, predictionlist_new)
+
+
+def relabel_image_cle(label_image, predictionlist):
+    """
+    Returns a label image where each label value corresponds
+    to the cluster identity defined by the predictionlist.
+    it is assumed that len(predictionlist) == max(label_image)
+
+    Parameters
+    ----------
+    label_image: ndarray or dask array
+        Label image used for cluster predictions
+    predictionlist: array
+        Array containing cluster identities for each label
+    """
+    import pyclesperanto_prototype as cle
```

Comment:
This code looks very similar to this section in napari-skimage-regionprops (nsr): https://github.com/haesleinhuepf/napari-skimage-regionprops/blob/master/napari_skimage_regionprops/_parametric_images.py#L23-L38

Do you think it would be possible to call that function from nsr instead of duplicating the code?
```
from napari_skimage_regionprops import relabel
relabel(image, predictionlist)
```

[Read more about pros and cons of code duplication](https://en.wikipedia.org/wiki/Duplicate_code)

Reply:
@haesleinhuepf,
Sure! I'm not sure why I didn't think of that. The holidays must have brought me back into the old code duplication habbit :D